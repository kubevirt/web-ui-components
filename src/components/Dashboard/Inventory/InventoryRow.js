import React from 'react';
import PropTypes from 'prop-types';

import { prefixedId } from '../../../utils';

import { InventoryItemStatus } from './InventoryItemStatus';
import { InlineLoading } from '../../Loading';

const InventoryRowTitle = ({ title, count }) => (
  <div className="kubevirt-inventory__row-title">{count != null ? `${count} ${title}` : title}</div>
);

InventoryRowTitle.defaultProps = {
  count: null,
};

InventoryRowTitle.propTypes = {
  title: PropTypes.string.isRequired,
  count: PropTypes.number,
};

export const InventoryRow = ({ title, count, LoadingComponent, ...rest }) => (
  <div id={prefixedId('inventory', title.toLowerCase())} className="kubevirt-inventory__row">
    <InventoryRowTitle count={count} title={title} />
    {count != null ? <InventoryItemStatus {...rest} key="status" /> : <LoadingComponent />}
  </div>
);

InventoryRow.defaultProps = {
  ...InventoryRowTitle.defaultProps,
  ...InventoryItemStatus.defaultProps,
  LoadingComponent: InlineLoading,
};

InventoryRow.propTypes = {
  ...InventoryRowTitle.propTypes,
  ...InventoryItemStatus.propTypes,
  LoadingComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
};
