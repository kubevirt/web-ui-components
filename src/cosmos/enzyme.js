import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
