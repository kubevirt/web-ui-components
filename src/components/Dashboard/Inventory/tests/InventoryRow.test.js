import React from 'react';
import { render } from 'enzyme';

import { InventoryRow } from '../InventoryRow';
import { default as InventoryRowFixtures } from '../fixtures/InventoryRow.fixture';

// eslint-disable-next-line react/prop-types
const testInventoryRow = ({ props }) => <InventoryRow {...props} />;

describe('<InventoryRow />', () => {
  InventoryRowFixtures.forEach(fixture => {
    it(`renders ${fixture.name} correctly`, () => {
      const component = render(testInventoryRow(fixture));
      expect(component).toMatchSnapshot();
    });
  });
});
