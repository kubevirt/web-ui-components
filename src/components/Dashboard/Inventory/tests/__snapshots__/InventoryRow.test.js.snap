// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InventoryRow /> renders correctly 1`] = `
<div
  class="kubevirt-inventory__row"
  id="inventory-pvcs"
>
  <div
    class="kubevirt-inventory__row-title"
  >
    5 PVCs
  </div>
  <div
    class="kubevirt-inventory__row-status"
  >
    <div
      class="kubevirt-inventory__row-status-item"
    >
      <span
        aria-hidden="true"
        class="pficon pficon-in-progress kubevirt-inventory__row-status-item-icon--in-progress kubevirt-dashboard__icon-sm"
      />
      <span
        class="kubevirt-inventory__row-status-item-text"
      >
        1
      </span>
    </div>
    <div
      class="kubevirt-inventory__row-status-item"
    >
      <span
        aria-hidden="true"
        class="fa fa-exclamation-circle kubevirt-inventory__row-status-item-icon--error kubevirt-dashboard__icon-sm"
      />
      <span
        class="kubevirt-inventory__row-status-item-text"
      >
        2
      </span>
    </div>
    <div
      class="kubevirt-inventory__row-status-item"
    >
      <span
        aria-hidden="true"
        class="fa fa-exclamation-triangle kubevirt-inventory__row-status-item-icon--warn kubevirt-dashboard__icon-sm"
      />
      <span
        class="kubevirt-inventory__row-status-item-text"
      >
        3
      </span>
    </div>
    <div
      class="kubevirt-inventory__row-status-item"
    >
      <span
        aria-hidden="true"
        class="pficon pficon-off kubevirt-inventory__row-status-item-icon--off"
      />
      <span
        class="kubevirt-inventory__row-status-item-text"
      >
        3
      </span>
    </div>
    <div
      class="kubevirt-inventory__row-status-item"
    >
      <span
        aria-hidden="true"
        class="fa fa-check-circle kubevirt-inventory__row-status-item-icon--ok kubevirt-dashboard__icon-sm"
      />
      <span
        class="kubevirt-inventory__row-status-item-text"
      >
        5
      </span>
    </div>
  </div>
</div>
`;
